### ISSUE TITLE
AffineTransform ScaleX and ScaleY return invalid values.

### ISSUE THREAD

**Issue by vpenades** [dotnet/Microsoft.Maui.Graphics#184]
AffineTransform's ScaleX and ScaleY are defined like:

```c#
public float ScaleX => _m00;
public float ScaleY => _m11;
```

These definitions will only return valid scales when the transform has no rotation at all (when _m01 and _m10 are zero), for all other cases, when it has some degree of rotation, ScaleX and ScaleY will return garbage.

A better implementation of ScaleX and ScaleY, that is true for all cases, would be this:

```c#
public float ScaleX => _m10 == 0 ? _m00 : new Vector2(_m00,_m10).Length();
public float ScaleY => _m01 == 0 ? _m11 : new Vector2(_m01,_m11).Length();
```

or just:

```c#
public float ScaleX => new Vector2(_m00,_m10).Length();
public float ScaleY => new Vector2(_m01,_m11).Length();
```

A generalization of deconstructing an AffineTransform as long as it's orthogonal:

```c#
public void Deconstruct(out Vector2 scale, out float rotation, out Vector2 translation)
        {         
            var sx = new Vector2(_m00,_m10).Length();
            var sy = new Vector2(_m01,_m11).Length();

            scale = new Vector2(sx, sy);

            rotation = (float)Math.Atan2(_m10, _m00);
            translation = new Vector2(_m02, _m12);
        }
```

**Comment by jonlipsky** [comment]
@vpenades You are correct that this could be misleading, and I'm happy to accept a pull request to change the behavior.

Like most things you've commented on, things are implemented the way they are because of specific use cases that I had, and I wasn't designing for anything beyond what I needed.  

As an FYI, this code pre-dates Skia, so it's not based on SKMatrix.  If you're curious, it was based on Java2D.

**Comment by vpenades** [comment]
Okey, then I would suggest to rename the properties... looking at microsoft's matrices, we have

- System.Numerics.Matrix3x2
  - M11, M12, M21, M22, M31, M32
- [System.Windows.Media.Matrix](https://docs.microsoft.com/es-es/dotnet/api/system.windows.media.matrix)
  - M11, M12, M21, M22, OffsetX, OffsetY

I would suggest to follow Systen.Numerics.Matrix3x2, so I would rename both the internal fields and the public properties.

Then, I would add these extra properties:
- SizeF Scale { get; }
- PointF Translation {get;}

Translation follows the same naming convention as Matrix3x2.

Scale would return the values along the axes of the matrix, so it would return the right scales, regardless of the rotation,

let me know if you agree, or if you need to change something about the proposal, and I'll do the pull request.

**Comment by jonlipsky** [comment]
@vpenades I agree with your proposal.  I really appreciate your collaboration on this!

**Comment by vpenades** [comment]
Okey, I'll be working on the PR... I think I will also need to add some unit tests
